### PR TITLE
Align API-Testing-Guidelines with new Enhancements and Alignments in Errors 

### LINKED ISSUE #321: Enhancements and Alignments in Errors
**Problem description**
Two enhancements are proposed inder this issue:
- Normalize mandatory errors that are common to every CAMARA API.
- Define specific Errors for `phoneNumber` concept in the same fashion as done for `device` in MetaRelease Fall 24 for [device error responses](https://github.com/camaraproject/Commonalities/blob/main/documentation/API-design-guidelines.md#62-error-responses---device-object) 

**Possible evolution**

1) Normalize Mandatory Error status
- Add within section `6.1 Standardized use of CAMARA error responses` the following information:

Mandatory Errors to be documented in CAMARA API Spec YAML are the following:
- For event notifications flow, the ones defined in [notification-as-cloud-event.yaml](https://github.com/camaraproject/Commonalities/blob/main/artifacts/notification-as-cloud-event.yaml)
- For event subscriptions APIs, the ones defined in [event-subscription-template.yaml](https://github.com/camaraproject/Commonalities/blob/main/artifacts/camara-cloudevents/event-subscription-template.yaml)
- For the rest of APIs:
> Error status 401
> Error status 403
> Error status 429 TOO_MANY_REQUESTS (For rate limit control)

> NOTE: The rest of Error status defined in section 6.1. will apply depending on specific considerations within the given WG.

2) Define specific Errors for `phoneNumber` concept

Add a new section `6.3 Error Responses - phoneNumber` to detail error scenarios for phoneNumber

| **Case #** | **Description**                                                            | **Error status** |         **Error code**         | **Message example**                                      |
|:----------:|:---------------------------------------------------------------------------|:----------------:|:------------------------------:|:---------------------------------------------------------|
|     0      | The request body does not comply with the schema                           |       400        |        INVALID_ARGUMENT        | Request body does not comply with the schema.           |
|     1      | The phoneNumber provided is not supported by the implementation |       422        | UNSUPPORTED_PHONE_NUMBER | phoneNumber is required.                                 |
|     2      | phoneNumber is not found.                    |       404        |        PHONE_NUMBER_NOT_FOUND        | phoneNumber not found.                             |  
|     3      | Invalid access token context                                               |       403        |     INVALID_TOKEN_CONTEXT      | phoneNumber is not consistent with access token. |
|     4      | Service not applicable to the phoneNumber                                    |       422        |     PHONE_NUMBER_NOT_APPLICABLE      | The service is not available for the provided phoneNumber.    |
|     5      | The phone number is not included in the request and the phone number information cannot be derived from the 3-legged access token |       422        |     UNIDENTIFIABLE_PHONE_NUMBER      | The phoneNumber cannot be identified. |
|     6      | The phoneNumber is unnecesarily included in a 3-legged scenario where the subscription is identified from the 3-legged access token |      422        |      UNNECESSARY_PHONE_NUMBER        |  The phoneNumber is unnecesarily provided

NOTE: New exceptions would be added to `CAMARA_common.yaml` for reference

**Alternative solution**

**Additional context**

There are CAMARA APIs where phoneNumber represents the end's user subscription (e.g SIM-SWAP, Number Verification, Carrier Billing, KYC family,...). Therefore having specific errors related to the concept of phoneNumber is the natural way to address error scenarios in a fully semantic and user-friendly way

Proposed as a child issue of #306

### PR DESCRIPTION
**Pull request by jlurien**
#### What type of PR is this?

* documentation
* tests

#### What this PR does / why we need it:

Update wording with consensus in #321 

#### Which issue(s) this PR fixes:

Fixes #321

#### Does this PR introduce a breaking change?

-  No

#### Changelog input

```
- API testing guidelines aligned with new enhancements in error management

```

### CODE REVIEW

File: documentation/API-Testing-Guidelines.md [hunk 1]
```diff
@@ -78,7 +78,9 @@ For the explicit subscriptions model:
 
 ### Error scenarios
 
-* All errors explicitly documented in the API spec must be covered by one or more dedicated error scenarios.
+* Test plan must be aligned with API specification regarding the coverage of HTTP statuses in the responses.
+  - All errors explicitly documented in the API spec must be covered by one or more dedicated error scenarios.
+  - On the contrary, HTTP Statuses which are not explicitly documented in the API specification should not be covered in the test plan. If any of the non-mandatory "status" at Commonalities level is relevant in the context of a given API and deserves to have a specific test, it must be documented as well in the API specification.
```
Comment (rartych) [inline documentation/API-Testing-Guidelines.md:83]: I propose to rephrase a bit, and directly reference to Design Guidelines:

```suggestion
  - On the contrary, HTTP Statuses which are not explicitly documented in the API specification should not be covered in the test plan. If any of the non-mandatory "status" in API Design Guidelines is relevant in the context of a given API and deserves to have a specific test, it needs to be documented in the API specification as well.
```

**Review by patrice-conil — APPROVED** [review]
LGTM

**Review by patrice-conil — APPROVED** [review]
LGTM

**Review by PedroDiez — APPROVED** [review]
LGTM

**Review by rartych — APPROVED** [review]

**Pull request merged**